### PR TITLE
Adding the Memory Allocation Optimizer

### PR DESCRIPTION
**Pull request by saxenism**

### CODE REVIEW

File: man/opt_common_subexpr.Rd [hunk 1]
```diff
@@ -26,4 +26,12 @@ code <- paste(
   sep = "\n"
```
Comment (jcrodriguez1989) [inline man/opt_common_subexpr.Rd]: there are several `man/` files that should not be included in this PR.

Comment (saxenism) [reply man/opt_common_subexpr.Rd]: Hmm, maybe the branch I created from your `develop` branch used an old `remote`. Whatever be the case, I'll look into it.

**Review by jcrodriguez1989 — APPROVED** [review]
Great work Rahul!
